src/api/rpc_api/author.rs

```diff
@@ -276,29 +282,34 @@ where
 		if watch_until < XtStatus::InBlock {
 			return Ok(report)
 		}
+		self.populate_events(&mut report).await?;
+		return Ok(report);
+	}
+
+	async fn populate_events(&self, report: &mut ExtrinsicReport<Self::Hash>) -> Result<()> {
```

Comment:
We can do it like this, but I think I would leave the error handling as it is currently. I see two problems:
- The report is not guaranteed to contain the events and then I feel a method `is_success(..)` is confusing as it cannot really check
- The `RawEventDetails` don't have access to the metadata needed to figure out the `DispatchError`
  - `fn get_associated_dispatch_error(&self, metadata: &Metadata)`
  - We can pass it along but it seems not so elegant

Reply:
Yes, that's true. In hindsight, it would have been better to use two different structs - one with events and one without. Might be worth the breaking change later on.

We can leave it as is, but what I really do not like, is that the function returns en Error even if the main functionality (at least as the function name suggests) of populating events succeeds. 

Would it make sense to differentiate here / split the function into two?

Reply:
Yes I agree the error handling is a bit awkward. I moved the error reporting to a separate function and like it much better. I'm not so sure about the name (`status_based_on_events`). Do you have a better idea?

Reply:
Yes, now it's much more understandable. Thank you very much!
One small challenge left: Since the error is now returned only if there are no events added to the report, would it make sense to return a new `ExtrinsicReport` in the Result instead of mutating the existing one?

Regarding the function name: how about `check_events_for_dispatch_error` , `screen_events`, `verify_events_for_onchain_success`
... or mabye any combination of those. I'm fine with all of them.

Reply:
I also thought about returning a new `ExtrinsicReport` instead of modification but decided against it because the modified object only contains additional data (nothing is lost). Otherwise we always have to create copy that is not needed in most cases (and the user still has the option to manually create a copy)

I renamed the function to `check_events_for_dispatch_error`.